kratos/tests/cpp_tests/strategies/schemes/test_residual_based_adjoint_bossak_scheme.cpp

```diff
@@ -451,6 +556,178 @@ class AdjointElement : public Element
+    typedef Kratos::unique_ptr<AdjointCondition> UniquePointer;
+
+
+    static Pointer Create(Node<3>::Pointer pNode1)
+    {
+        auto nodes = PointerVector<Node<3>>{};
+        nodes.push_back(pNode1);
+        return Kratos::make_intrusive<AdjointCondition>(nodes);
+    }
+
+    AdjointCondition(const NodesArrayType& ThisNodes)
+        : Condition(0, ThisNodes), mPrimalCondition(ThisNodes)
+    {
+        SetValue(ADJOINT_EXTENSIONS, Kratos::make_shared<ThisExtensions>(this));
+    }
+
+    void EquationIdVector(EquationIdVectorType& rResult, const ProcessInfo& rCurrentProcessInfo) const override
+    {
+        rResult.resize(1);
+        rResult[0] = this->GetGeometry()[0].GetDof(ADJOINT_VECTOR_1_X).EquationId();
+    }
+
+    void GetDofList(DofsVectorType& rConditionalDofList, const ProcessInfo& rCurrentProcessInfo) const override
+    {
+        rConditionalDofList.resize(1);
+        rConditionalDofList[0] = this->GetGeometry()[0].pGetDof(ADJOINT_VECTOR_1_X);
+    }
+
+    void GetValuesVector(Vector& values, int Step = 0) const override
+    {
+        values.resize(1);
+        values[0] = this->GetGeometry()[0].FastGetSolutionStepValue(ADJOINT_VECTOR_1_X, Step);
+    }
+
+    void GetFirstDerivativesVector(Vector& values, int Step = 0) const override
+    {
+        values.resize(1);
+        values[0] = this->GetGeometry()[0].FastGetSolutionStepValue(ADJOINT_VECTOR_2_X, Step);
+    }
+
+    void GetSecondDerivativesVector(Vector& values, int Step = 0) const override
+    {
+        values.resize(1);
+        values[0] = this->GetGeometry()[0].FastGetSolutionStepValue(ADJOINT_VECTOR_3_X, Step);
+    }
+
+    void CalculateLeftHandSide(MatrixType& rLeftHandSideMatrix,
+                               const ProcessInfo& rCurrentProcessInfo) override
+    {
+        mPrimalCondition.CalculateLeftHandSide(rLeftHandSideMatrix, rCurrentProcessInfo);
+        noalias(rLeftHandSideMatrix) = -rLeftHandSideMatrix;
+    }
+
+    void CalculateFirstDerivativesLHS(MatrixType& rLeftHandSideMatrix,
+                                      const ProcessInfo& rCurrentProcessInfo) override
+    {
+        KRATOS_TRY;
+        mPrimalCondition.CalculateDampingMatrix(rLeftHandSideMatrix, rCurrentProcessInfo);
+        noalias(rLeftHandSideMatrix) = -rLeftHandSideMatrix;
+        KRATOS_CATCH("");
+    }
+
+    void CalculateSecondDerivativesLHS(MatrixType& rLeftHandSideMatrix,
+                                       const ProcessInfo& rCurrentProcessInfo) override
+    {
+        KRATOS_TRY;
+        mPrimalCondition.CalculateMassMatrix(rLeftHandSideMatrix, rCurrentProcessInfo);
+        noalias(rLeftHandSideMatrix) = -rLeftHandSideMatrix;
+        KRATOS_CATCH("");
+    }
+
+    void CalculateSensitivityMatrix(const Variable<double>& rDesignVariable,
+                                    Matrix& rOutput,
+                                    const ProcessInfo& rCurrentProcessInfo) override
+    {
```

Comment:
Here you should discuss that this function will be not called by the sensitivity builder

Reply:
done :)